classes/auth.php

```diff
@@ -676,7 +676,7 @@ public function saml_login_complete($attributes) {
                 $this->update_user_record_from_attribute_map($user, $attributes, true);
                 if (empty($user->username)) {
                     // Just in case username field not set, use uid.
-                    $user->username = $uid;
+                    $user->username = strtolower($uid);
```

Comment:
I had a quick look Dan , it _seems_ fine. Small concern whether or not the check prior (e.g. to see if the user exists before creation) uses the uid as is, or does it with various saml2_settings for different case sensitivities which might mean after the initial creation, the next one might fail if that has been configured differently. (e.g. the call around line 636)

Otherwise if that is okay, I'm happy for this to merge.